### PR TITLE
Casey multiple experiments

### PR DESCRIPTION
**Pull request by sometimescasey**
Added a bunch of helpers to run training for all the signatures and save weights/experiment results. Will run this today @ 250 epochs each on about 60 sigs.

Refactored `vanilla_vae.py` and `experiments.py` to be a little more OO/modularized. Let me know if anything gravely offends anyone. I shouldn't have broken anything else you guys have created.

### CODE REVIEW

File: cmiyc/experiments.py [hunk 1]
```diff
@@ -4,73 +4,141 @@
...
+    # TODO: check these
+    args = {
+        'classifier': 'forest',
+        'sig_id':     1,
+        'image_res':  128,
+        'intermediate_dim': 512,
+        'latent_dim': 256,
+        'save_dir': 'saved-models/models_res128_id512_ld256_epoch250.h5'
+    }
+
```
Comment (PsiPhiTheta) [inline cmiyc/experiments.py:181]: I agree this is cleaner. As long as it compiles and runs fine, its OK by me

**Review by PsiPhiTheta — APPROVED** [review]
Haven't run this locally yet but the reorganisation makes sense to me

**Review by PsiPhiTheta — APPROVED** [review]
Also agree with these changes, will check if they run nicely soon

**Comment by sometimescasey** [comment]
I've tested quite alot, everything runs fine off their respective `__main__` on my end but happy to wait till you take a look. No rush.

**Review by PsiPhiTheta — APPROVED** [review]
I don't know how the unique sig IDs work but I trust you. How does it work in the end btw?

**Review by PsiPhiTheta — APPROVED** [review]
Seems fine to me yep

**Review by PsiPhiTheta — APPROVED** [review]
Yep np

**Comment by PsiPhiTheta** [comment]
> I've tested quite alot, everything runs fine off their respective `__main__` on my end but happy to wait till you take a look. No rush.

No worries, it looks fine to me and we can always revert if we have to but I doubt it will be required

**Comment by sometimescasey** [comment]
> I don't know how the unique sig IDs work but I trust you. How does it work in the end btw?

Oh I think I misunderstood your q. I'm literally just taking the entire training data dataframe and reading all the unique values from the `sig_id` column.

**Pull request merged**